### PR TITLE
🖍 Interactive results redesign

### LINKED ISSUE #29793: [Story interactives] Add new designs of results components
The components were redesigned to account for categories without images to look good as well. Follow the image below
image

### PR DESCRIPTION
**Pull request by mszylkowski**
Redesigned the results component in a way that is extensively customizable and more adaptive to lack of images/descriptions.

This new design accounts for:
- Image vs no image: When there's no image (and it's not transparent) the score bar takes the prompt background color to add color to the component. The image container is removed. This means the component will look good without an image (even though it's encouraged).
- Score vs no score: When there's no score, it adapts the gap at the top to balance the elements inside the component better.
- Prompt background: If the prompt background (score background) is transparent (because there is an image or because the var(--interactive-prompt-background) is transparent), the margins get adjusted. The prompt background can be set to low opacity for cool effects on the transparent theme, but defaults to transparent. If light/dark themes with no image and transparent prompt, the score row won't have a background, then the margins get adjusted properly as well.

Percentage symbol was added to let users know better what the number is referring to.

All the examples below are implemented in the results example story, and can be written with valid AMP.

<table>
<tr>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
</tr>
</table>

Closes #29793

### CODE REVIEW

**Comment by mszylkowski** [comment]
@processprocess not sure why Github requested your review again, didn't change anything significant so no need to review again 😂

File: extensions/amp-story-interactive/1.0/amp-story-interactive-results.css [hunk 1]
```diff
@@ -25,26 +25,43 @@
...
+
+.i-amphtml-story-interactive-results-container.i-amphtml-story-interactive-has-image:not(.i-amphtml-story-interactive-has-score) {
+  padding-top: 40px !important;
+}
+
+.i-amphtml-story-interactive-has-score .i-amphtml-story-interactive-results-top {
+  display: flex !important;
+  align-items: center !important;
+  justify-content: space-between !important;
+  padding: 15px 20px !important;
```
Comment (gmajoulet) [inline extensions/amp-story-interactive/1.0/amp-story-interactive-results.css:39]: Here and above, are you using padding as a margin?

File: extensions/amp-story-interactive/1.0/amp-story-interactive-results.js [hunk 2]
```diff
@@ -42,16 +43,15 @@ const buildResultsTemplate = (element) => {
...
       <div class="i-amphtml-story-interactive-results-top">
-        <div class="i-amphtml-story-interactive-results-top-score">SCORE</div>
-        <div class="i-amphtml-story-interactive-results-top-line"></div>
-        <div class="i-amphtml-story-interactive-results-top-value">100</div>
-      </div>
-      <div class="i-amphtml-story-interactive-results-visuals">
-        <div class="i-amphtml-story-interactive-results-dots"></div>
-        <div class="i-amphtml-story-interactive-results-image-border">
-          <div class="i-amphtml-story-interactive-results-image"></div>
+        <div class="i-amphtml-story-interactive-results-top-score">SCORE:</div>
```
Comment (gmajoulet) [inline extensions/amp-story-interactive/1.0/amp-story-interactive-results.js:46]: Please file a follow up ticket to translate this string

File: extensions/amp-story-interactive/1.0/amp-story-interactive-results.js [hunk 3]
```diff
@@ -264,4 +276,22 @@ export class AmpStoryInteractiveResults extends AmpStoryInteractive {
   updateStoryStoreState_(unusedOption) {
     // Prevent from updating the state.
   }
+
+  /**
+   * Check score background has a color with alpha 0, used to adjust layout
+   * @return {bool}
```
Comment (gmajoulet) [inline extensions/amp-story-interactive/1.0/amp-story-interactive-results.js:282]:
```suggestion
   * @return {boolean}
```

File: extensions/amp-story-interactive/1.0/amp-story-interactive-results.js [hunk 4]
```diff
@@ -264,4 +276,22 @@ export class AmpStoryInteractiveResults extends AmpStoryInteractive {
...
+   * @private
+   **/
+  scoreBackgroundIsTransparent_() {
```
Comment (gmajoulet) [inline extensions/amp-story-interactive/1.0/amp-story-interactive-results.js:285]: Why can't you rely on the `chip-style=transparent` attribute?

**Pull request merged**